### PR TITLE
store read infos into model + display "..." when name is too long

### PR DESCRIPTION
**Pull request by ivandrofly**
- read dictionary info from zip to model
- when name is too long to be displayed on combobox-dictionary suffix it with "..." 

image

### CODE REVIEW

File: src/ui/Forms/GetDictionaries.cs [hunk 1]
```diff
@@ -12,9 +13,33 @@ namespace Nikse.SubtitleEdit.Forms
...
+            /// Native Name
+            /// </summary>
+            public string NativeName { get; set; }
+            public string Description { get; set; }
+            public string DownloadLink { get; set; }
+
+            public override string ToString()
+            {
+                var displayName = $"{Name}{(string.IsNullOrEmpty(NativeName) ? "" : $" - {NativeName}")}";
+                if (displayName.Length > 55)
```
Comment (niksedk) [inline src/ui/Forms/GetDictionaries.cs:32]: Could you use `comboBoxDictionaries.DropDownWidth` instead of 55?

Comment (ivandrofly) [reply src/ui/Forms/GetDictionaries.cs:32]: `comboBoxDictionaries.Width or comboBoxDictionaries.DropDownWidth` is not working properly.

I also tried to use `TextWiddth.CalcPixelWidth` but there are way too different in value when compared with `comboBoxDictionaries.DropDownWidth`

File: src/ui/Forms/GetDictionaries.cs [hunk 2]
```diff
@@ -108,7 +105,30 @@ private void LoadDictionaryList(string xmlResourceName)
...
+                        if (width > comboboxWidth)
+                        {
+                            double pixelChar = width / text.Length;
+                            var charCount = (int)Math.Floor(comboboxWidth / pixelChar);
+                            displayText = text.Substring(0, charCount - 3) + "...";
+                        }
+
+                        dictionaryItem.DisplayText = displayText;
+                    }
+                }
```
Comment (ivandrofly) [inline src/ui/Forms/GetDictionaries.cs:128]: @niksedk updated

File: src/ui/Forms/GetDictionaries.cs [hunk 3]
```diff
@@ -77,18 +96,40 @@ private void LoadDictionaryList(string xmlResourceName)
...
+                            Description = description,
+                            DownloadLink = downloadLink
+                        });
+                    }
+                }
+
+                // ensure ellipses suffix on text overlaps
+                using (var graphics = Graphics.FromHwnd(IntPtr.Zero))
+                {
+                    double comboboxWidth = comboBoxDictionaries.Width;
```
Comment (niksedk) [inline src/ui/Forms/GetDictionaries.cs:112]: Should probably be DropDownWidth, right?

Comment (ivandrofly) [reply src/ui/Forms/GetDictionaries.cs]: > Should probably be DropDownWidth, right?

Changed :) but they return the same value!

**Pull request merged**